pennylane_orquestra/cli_actions.py

```diff
@@ -0,0 +1,236 @@
+            workflow_id = res[1].split()[-1]
+        except IndexError as e:
+            raise ValueError(unexpected_resp_msg) from e
+    else:
+        raise ValueError(unexpected_resp_msg)
+
+    return workflow_id
+
+
+def workflow_details(workflow_id):
+    """Function for getting workflow details via a CLI call.
+
+    Args:
+        workflow_id (str): the workflow id for which information will be
+            retrieved
+
+    Returns:
+        list: the list of strings representing the response message of the CLI
+        call
+    """
+    return qe_get(workflow_id, option="workflow")
+
+
+def workflow_results(workflow_id):
+    """Function for getting workflow results via a CLI call.
+
+    Args:
+        workflow_id (str): the workflow id for which information will be
+            retrieved
+
+    Kwargs:
+        option (str): The option specified for the ``qe get`` CLI call.
+            Examples include ``workflow`` and ``workflowresult``.
+
+    Returns:
+        list: the list of strings representing the response message of the CLI
+        call
+    """
+    return qe_get(workflow_id, option="workflowresult")
+
+
+def write_workflow_file(filename, workflow):
+    """Write a workflow file given the name of the file.
+
+    This function will create a YAML file with the workflow content. The file
+    is placed into a user-specific data folder specified by using
+    ``appdirs.user_data_dir``.
+
+    Args:
+        filename (str): the name of the file to write
+        workflow (dict): the workflow generated as a dictionary
+    """
+    # Get the directory to write the file to
+    directory = user_data_dir("pennylane-orquestra", "Xanadu")
+
+    # Create target Directory if it doesn't exist
+    os.makedirs(directory, exist_ok=True)
+
+    filepath = os.path.join(directory, filename)
+
+    with open(filepath, "w") as file:
+        # The order of the keys within the YAML file is pre-defined for Orquestra,
+        # hence need to keep the order
+        yaml.dump(workflow, file, sort_keys=False)
+
+    return filepath
+
+
+def loop_until_finished(workflow_id, timeout=300):
+    """Loops until the workflow execution has finished by querying workflow
+    details using the workflow ID.
+
+    The flows of messages and the checks were based on responses obtained when
+    using Orquestra API v1.0.0.
+
+    Args:
+        workflow_id (str): the ID of the workflow for which to return the
+            results
+
+    Keyword args:
+        timeout (int): seconds to wait until raising a TimeoutError
+
+    Returns:
+        dict: the resulting dictionary parsed from a json file
+    """
+    start = time.time()
+    query_results = True
+    tries = 0
+    while query_results:
+        tries += 1
+
+        # Check if we've exceeded the timeout time, otherwise loop further
+        if time.time() - start > timeout:
+            current_status = workflow_details(workflow_id)
+            raise TimeoutError(
```

Comment:
I didn't know about timeout error! nice

Reply:
Oh was not documented, good point! Added to docstring